src/cmd-buildextend-metal

```diff
@@ -118,26 +113,43 @@ if [ "${rev_parsed}" != "${commit}" ]; then
     fi # otherwise, the repo already has a ref, so no need to create
 fi
 
+
+itype=metal
+img=${name}-${build}-${itype}.raw
+if [ -f "${builddir}/${img}" ]; then
+    echo "Bare metal image already exists"
+    exit
+fi
+
+path=${PWD}/${img}
 # for anaconda logs
 mkdir -p tmp/anaconda
 
-for itype in ${BIOS:+metal-bios} ${UEFI:+metal-uefi}; do
-    img=${name}-${build}-${itype}.raw
-    if [ -f "${builddir}/${img}" ]; then
-        echo "Image $itype already exists"
-        continue
-    fi
-
-    path=${PWD}/${img}
-    run_virtinstall "${ostree_repo}" "${ref}" "${path}.tmp" --variant="${itype}"
+if [ "$arch" == "aarch64" ]; then
+    run_virtinstall "${ostree_repo}" "${ref}" "${path}.tmp" --variant="${itype}-uefi"
     /usr/lib/coreos-assembler/gf-platformid "${path}"{.tmp,} metal
     rm -f "${path}".tmp
+else
+    if [ ! -e "$PWD/tmp/ostree-size.json" ]; then
+        /usr/lib/coreos-assembler/estimate-commit-disk-size --repo "$ostree_repo" "$ref" > "$PWD/tmp/ostree-size.json"
```

Comment:
An `echo "Estimating disk size..."` here would be useful, since it does take some time.